common/flatpak-dir.c

```diff
@@ -1935,6 +1935,24 @@ flatpak_deploy_data_get_eol_rebase (GVariant *deploy_data)
   return eol;
 }
 
+const char **
+flatpak_deploy_data_get_previous_ids (GVariant *deploy_data, gsize *length)
+{
+  g_autoptr(GVariant) metadata = g_variant_get_child_value (deploy_data, 4);
+  g_autoptr(GVariant) previous_ids_v = NULL;
+  const char **previous_ids = NULL;
+
+  previous_ids_v = g_variant_lookup_value (metadata, "previous-ids", NULL);
+  if (previous_ids_v)
+    {
+      previous_ids = g_variant_get_strv (previous_ids_v, NULL);
+      if (length != NULL)
+        *length = g_strv_length ((char **) previous_ids);
+    }
```

Comment:
You should also have:
```
else
  {
    if (length != NULL)
      *length = 0;
  }
```

so that `length` is never left undefined.

Reply:
can't we : 
```
else if (length != NULL)
  *length = 0;
```
I think we can, but your code block is more read-able I think :)